packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -252,6 +263,66 @@ class TextMapPropagator {
     carrier.tracestate = ts.toString()
   }
 
+  _injectAwsXrayContext (spanContext, carrier) {
+    // injects AWS Trace Header (X-Amzn-Trace-Id) to carrier
+    // ex: 'Root=1-00000000-00000000fffffffffffffffe;Parent=ffffffffffffffff;Sampled=1;_dd.origin=fakeOrigin;
+
+    // based off: https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader
+    if (!this._hasPropagationStyle('inject', 'xray')) return
+
+    // TODO: Do we have access to a start time? Java does
+    const e2eStart = this._getEndToEndStartTime(spanContext.start_ms)
+
+    let str = (
+      xrayRootKey + '=' +
+      xrayRootPrefix +
+      e2eStart + '-' +
+      spanContext._traceId.toString().padStart(24, '0') +
+      ';' + xrayParentKey + '=' +
+      spanContext._spanId.toString().padStart(16, '0')
+    )
+
+    if (spanContext?._sampling?.priority) {
+      str += ';' + xraySampledKey + '=' + (spanContext._sampling.priority > 0 ? '1' : '0')
+    }
+
+    const maxAdditionalCapacity = xrayMaxAdditionalBytes - str.length
+
+    const origin = spanContext._trace.origin
+    if (origin) {
+      this._addXrayBaggage(str, xrayOriginKey, origin, maxAdditionalCapacity)
+    }
+    if (e2eStart !== '00000000') {
+      this._addXrayBaggage(str, xrayE2EStartTimeKey, e2eStart.toString(), maxAdditionalCapacity)
+    }
+
+    for (const [key, value] of Object.entries(spanContext._baggageItems)) {
+      if (!this._isReservedXrayKey(key)) {
+        str = this._addXrayBaggage(str, key, value, maxAdditionalCapacity)
+      }
+    }
+
+    carrier[xrayHeaderKey] = str
+  }
+
+  _getEndToEndStartTime (start) {
+    if (!start) return '00000000'
```

Comment:
Would it make sense to replace the two `'00000000'` entries with something like `const emptyTime = '00000000'`?

Reply:
done.